src/blocking_rx.rs

```diff
@@ -67,6 +68,13 @@ impl<T> Drop for Rx<T> {
     }
 }
 
+impl<T> From<AsyncRx<T>> for Rx<T> {
+    fn from(value: AsyncRx<T>) -> Self {
+        let value = ManuallyDrop::new(value);
```

Comment:
I suggest no unsafe code here;
you can call shared.add_rx() count and let the drop happen, since there might be other stuff to be added later